src/NumericalAlgorithms/Interpolation/Events/InterpolateWithoutInterpComponent.hpp

```diff
@@ -0,0 +1,189 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Domain/ElementId.hpp"
+#include "NumericalAlgorithms/Interpolation/Actions/InterpolationTargetVarsFromElement.hpp"
+#include "NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp"
+#include "NumericalAlgorithms/Interpolation/PointInfoTag.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Tags {
+struct TimeStepId;
+}  // namespace Tags
+namespace domain {
+namespace Tags {
+template <size_t VolumeDim>
+struct Mesh;
+}  // namespace Tags
+}  // namespace domain
+namespace Tags {
+template <typename TagsList>
+struct Variables;
+}  // namespace Tags
+template <size_t Dim>
+class Mesh;
+template <size_t VolumeDim>
+class ElementId;
+namespace intrp {
+template <typename Metavariables, typename Tag>
+struct InterpolationTarget;
+}  // namespace intrp
+/// \endcond
+
+namespace intrp {
+namespace Events {
+template <size_t VolumeDim, typename InterpolationTargetTag,
+          typename Metavariables, typename Tensors, typename EventRegistrars>
+class InterpolateWithoutInterpComponent;
+
+namespace Registrars {
+template <size_t VolumeDim, typename InterpolationTargetTag,
+          typename Metavariables, typename Tensors>
+struct InterpolateWithoutInterpComponent {
+  template <typename RegistrarList>
+  using f = Events::InterpolateWithoutInterpComponent<
+      VolumeDim, InterpolationTargetTag, Metavariables, Tensors, RegistrarList>;
+};
+}  // namespace Registrars
+
+/// Does an interpolation onto an InterpolationTargetTag by calling Actions on
+/// the InterpolationTarget component.
+template <size_t VolumeDim, typename InterpolationTargetTag,
```

Comment:
just move this forward declaration above to replace the other one

Reply:
Sorry, I don't understand what you mean.  Does "the other one" mean the forward declaration on line 48?  I don't understand how that can be replaced because Registrars::InterpolateWithoutInterpComponent depends on it (and the forward declaration on line 65 depends on Registrars::InterpolateWithoutInterpComponent).

Reply:
I'm surprised this works; I would forward declare `Registrars::InterpolateWithoutInterpComponent`, then this forward declaration, and then the definitions...